www/src/pages/blog/astro-019.astro

```diff
@@ -0,0 +1,171 @@
+    <style global>
+      img {
+        max-width: 100%;
+      }
+    </style>
+  </head>
+
+  <body>
+    <BlogHeader />
+    <BlogPost title={title} heroImage={heroImage} publishDate={publishDate}>
+        <Markdown>
+          We are excited to introduce Astro v0.19.0, featuring:
+
+          * __[Next.js-inspired routing:](#file-based-routing-inspired-by-nextjs)__ Your `pages/` file layout defines your routes.
+          * __[`client:only` directive:](#clientonly-loading-directive)__ Disable SSR for individual components.
+          * __[`Astro.resolve()`:](#astroresolve)__ Resolve relative URLs in component attributes.
+          * __[Community translations:](#docs-translations)__ Already available in 10 languages.
+          * __[Open Collective:](#open-collective)__ Now accepting sponsorship!
+
+          ## File-based routing, inspired by Next.js
+
+          Astro has always supported basic file-based routing, where every file in your `pages/` directory created a new page at that same URL. Inspired by [Next.js](https://nextjs.org/docs/routing/dynamic-routes) and [SvelteKit](https://kit.svelte.dev/docs#routing-pages), you can now add brackets to your page filename (ex: `src/pages/[param].astro`) to create a dynamic param in your final URL. With this new syntax, Astro adds support for URL params, slugs, pretty URLs, pagination and more.
+
+          To create a dynamic route, create a file in your pages directory like `pages/posts/[slug].astro`. Define a [`getStaticPaths()`](https://docs.astro.build/reference/api-reference#getstaticpaths) function and tell Astro which paths to build for the route:
+
+          ```astro
+          ---
+          // src/pages/posts/[slug].astro
+          // Tell Astro what pages to build for your route "/pages/:slug"
+          export async function getStaticPaths() {
+            return [
+              // Generates: /pages/hello-world
+              { params: { slug: 'hello-world' } },
+              // Generates: /pages/my-first-blog-post
+              { params: { slug: 'my-first-blog-post' } },
+              // Generates: /pages/astro-ftw
+              { params: { slug: 'astro-ftw' } },
+              // ...
+            ];
+          }
+          ---
+          /* your page HTML here! */
+          ```
+
+          Because `getStaticPaths()` is an async function, you can use it to load any external data that you might like. We normally love to use the [Pokemon API](https://pokeapi.co/) in our examples, but this system now works just as well with your favorite headless CMS.
+
+          ```js
+          // src/pages/posts/[id].astro
+          // Tell Astro what pages to build for your route "/pages/:id"
+          export async function getStaticPaths() {
+            // Lets fetch posts from a Wordpress CMS:
+            const CSS_TRICKS_CMS = 'https://css-tricks.com/wp-json/wp/v2/posts?per_page=12&_embed';
```

Comment:
I assume this is publicly known and we're not sharing a secret URL of theirs...?

Reply:
Oh, I should have kept reading down first, never mind